docs/monitoring.md

```diff
@@ -79,3 +79,21 @@ scrape_configs:
     action: replace
     target_label: kubernetes_name
 ```
+
+# Volume health monitoring
+
+This is a [CSI feature](https://kubernetes.io/docs/concepts/storage/volume-health-monitoring/) introduced as an Alpha feature in Kubernetes v1.19 and a second Alpha was done in v1.21. This feature is to detect "abnormal" volume conditions and report them as events on PVCs and Pods. A DirectPV volume will be considered as "abnormal" if the respective volume mounts are not present in the host.
+
+For node side monitoring, the feature gate `CSIVolumeHealth` needs to be enabled. However, DirectPV also installs external health monitor controller which monitors and reports volume health events to PVCs.
+
+To enable volume health monitoring, Install directpv with `--enable-volume-health-monitoring` flag.
+
+```sh
+kubectl directpv install --enable-volume-health-monitoring
+```
+
+For private registries, please note that the following image is required for enabling volume health monitoring
+
+```
+quay.io/minio/csi-external-health-monitor-controller:v0.10.0
```

Comment:
1. You would need to update https://github.com/minio/directpv/blob/master/docs/tools/push-images.sh instead of documenting here
2. You would need to update https://github.com/minio/directpv/blob/master/docs/command-reference.md and https://github.com/minio/directpv/blob/master/docs/installation.md#directpv-csi-driver-installation

Reply:
Volume health monitoring is not enabled by default, so, for a fresh installation we don't need this. Users need this image only when they enable volume health monitoring

Reply:
Users make common mistakes to miss pulling this image when enabling it. There is no problem to have unused image of `csi-external-health-monitor-controller`. When health monitoring is enabled, it works seamlessly than failures.